pyserini/encode/mbeir/mbeir_dataset.py

```diff
@@ -0,0 +1,58 @@
+
+# Pyserini: Reproducible IR research with sparse and dense representations
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from torch.utils.data import Dataset
+from PIL import Image
+
+
+class MBEIRCorpusDataset(Dataset):
+    def __init__(self, batch_info, img_preprocess_fn, **kwargs):
+        data = []
+        num_records = len(batch_info["did"])
+        for i in range(num_records):
+            record = {
+                "did": batch_info["did"][i],
+                "img_path": batch_info["img_path"][i],
+                "modality": batch_info["modality"][i],
+                "txt": batch_info["txt"][i],
+            }
+            data.append(record)
+        self.data = data
+        self.img_preprocess_fn = img_preprocess_fn
+        self.kwargs = kwargs
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, idx):
+        entry = self.data[idx]
+        img_path = entry.get("img_path", None)
+        if img_path is not None:
```

Comment:
isn't this enough to take care img_path None?
It is not clear to me why you need to make the changes to the fields and info handling logic